prow/crier/reporters/gcs/kubernetes/reporter.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"math"
+	"path"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/crier/reporters/gcs/internal/util"
+)
+
+const reporterName = "gcsk8sreporter"
+
+type gcsK8sReporter struct {
+	cfg            config.Getter
+	dryRun         bool
+	logger         *logrus.Entry
+	author         util.Author
+	rg             resourceGetter
+	reportFraction float32
+}
+
+type PodReport struct {
+	Pod    *v1.Pod    `json:"pod,omitempty"`
+	Events []v1.Event `json:"events,omitempty"`
+}
+
+type resourceGetter interface {
+	GetPod(cluster, namespace, name string) (*v1.Pod, error)
+	GetEvents(cluster, namespace string, pod *v1.Pod) ([]v1.Event, error)
+}
+
+type k8sResourceGetter struct {
+	podClientSets map[string]corev1.CoreV1Interface
+}
+
+func (rg k8sResourceGetter) GetPod(cluster, namespace, name string) (*v1.Pod, error) {
+	return rg.podClientSets[cluster].Pods(namespace).Get(name, metav1.GetOptions{})
+}
+
+func (rg k8sResourceGetter) GetEvents(cluster, namespace string, pod *v1.Pod) ([]v1.Event, error) {
+	events, err := rg.podClientSets[cluster].Events(namespace).Search(scheme.Scheme, pod)
+	if err != nil {
+		return nil, err
+	}
+	return events.Items, nil
+}
+
+func (gr *gcsK8sReporter) Report(pj *prowv1.ProwJob) ([]*prowv1.ProwJob, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second) // TODO: pass through a global context?
+	defer cancel()
+
+	_, _, err := util.GetJobDestination(gr.cfg, pj)
+	if err != nil {
+		gr.logger.Infof("Not uploading %q (%s#%s) because we couldn't find a destination: %v", pj.Name, pj.Spec.Job, pj.Status.BuildID, err)
```

Comment:
nit: Is this the right log level? If we don't expect all jobs to have a destination this should probably be debug and if we do expect all jobs to have a destination it should probably be a warning.